.github/workflows/default.yaml

```
@@ -109,10 +109,35 @@ jobs:
           sudo apt-get --quiet=2 build-dep squid
           sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin
 
+      - name: Install prerequisite MacOS packages
+        if: runner.os == 'macOS'
+        run: |
+          brew install \
+            automake coreutils cppunit gawk \
+            gnu-getopt gnu-sed grep libtool \
+            make openldap openssl@3 cyrus-sasl
+
       - name: Checkout sources
         uses: actions/checkout@v4
 
-      - run: ./test-builds.sh
+      - name: Run build on Linux
+        if: runner.os == 'Linux'
+        run: ./test-builds.sh
+
+      - name: Run build on MacOS
+        if: runner.os == 'macOS'
+        run: |
+          eval `brew shellenv`
+          export PKG_CONFIG_PATH="$HOMEBREW_PREFIX/lib/pkgconfig:$HOMEBREW_PREFIX/opt/openldap/lib/pkgconfig:$HOMEBREW_PREFIX/opt/cyrus-sasl/lib/pkgconfig"
+          export GETOPT="$HOMEBREW_PREFIX/opt/gnu-getopt/bin/getopt"
+          if gcc --version | grep -q clang ; then
+            export CFLAGS="-Wno-compound-token-split-by-macro${CFLAGS+ $CFLAGS}" # needed fir ltdl
```

```
${name+word}
${name:+word}
       If name is set, or in the second form is non-null, then
       substitute word; otherwise substitute nothing.
```

Don't we want the "second form" here (to avoid implying that we want to add a null/empty CFLAGS value)?

```suggestion
            export CFLAGS="-Wno-compound-token-split-by-macro${CFLAGS:+ $CFLAGS}" # needed fir ltdl
```

Added bonus: More folks might be familiar with `${x:+z}` notation than with `${x+z}`. FWIW, it took me a while to find documentation for the latter.

Same for the other two occurrences below, of course.

---

It is more future proof, in that it will handle CFLAGS lines being added earlier in the script cleanly, so there is some value.
Adopting the :+ syntax